modules/http/http-grammar.ym

```diff
@@ -109,6 +120,24 @@ http_options
     |
     ;
 
+on_error_action
+    : KW_SUCCESS { $$ = on_error_success; }
+    | KW_RETRY { $$ = on_error_retry; }
+    | KW_DROP { $$ = on_error_drop; }
+    | KW_DISCONNECT { $$ = on_error_disconnect; }
+    ;
+
+on_error_block
+    : '(' { last_on_error_params = g_new0(OnErrorParams, 1); }
```

Comment:
I would try to avoid allocation here, as syntax errors would generate leaks.

Reply:
Fixed.